### ISSUE TITLE
How to obtain coefficient of PDE using finite differences and DEEPXDE 

### ISSUE THREAD

**Issue by axledos100** [lululxvi/deepxde#114]
Hello Lu Lu, 

I came across an article on Medium.com (attached below): The neural network approach to solving inverse problems for physics, and was wonder how I could use DEEPXDE to solve this problem. Suppose I have a transport equation u_t + au_x,  and I am provided the solution data to this PDE, which has been approximated using finite-differences. Moreover, suppose I would like to solve the advection speed, a. How may I use Deepxde/PINNs to solve for the advection speed by having Deepxde/PINNs provide the advection speed, then the user solves the transport by finite difference using the neural network provided a, then measure the error with the solution data, then backpropagate? Is this possible to do using Deepxde?  

Thank you, 
Axle

Article Link: https://medium.com/@kailaix16/the-neural-network-approach-to-solving-inverse-problems-for-physics-1b3177f11872

**Comment by lululxvi** [comment]
It seems that the only difference between this inverse problem and the inverse problems in DeepXDE paper is that here `a` is a function of `x`. This can be easily done in DeepXDE, see examples in the [paper](https://www.osapublishing.org/oe/abstract.cfm?uri=oe-28-8-11618). The way to do it is that the network has two outputs: the first one is the solution `u`, and the second on is `a`, and then construct the loss as usual.

**Comment by axledos100** [comment]
Hello Lu Lu,

Thank you from your prompt response. I will read the paper tomorrow and attach the attempted code if I run into issues. I'm enjoying using Deepxde and it's inversion capabilities. Great work!

**Comment by axledos100** [comment]
Hello Lu Lu,

Is there any code from the paper that I may reference?

Best, 
Axle

**Comment by lululxvi** [comment]
You may send an email to the first author to ask for the code.

**Comment by axledos100** [comment]
Hello Lu Lu,
 
Sounds good! Do would you mind providing his official email? 

Thank you,
Axle

**Comment by lululxvi** [comment]
It is yuyao at bu dot edu

**Comment by axledos100** [comment]
Hello Lu Lu, 

I wrote to Yuyao a few days back, and he has still not gotten back to me. Is there any way you may contact him, please?
If you prefer discussing this in an email, I'd be more than happy to. 

Best, 
Axle

**Comment by lululxvi** [comment]
Actually the implementation is not hard. I can guide you. For example, the PDE is `u_t + au_x = 0`, here `u(x, t)` is a function of `x` and `t`, and `a(x, t)` is also an unknown function of `x` and `t`. (If `a` is an unknown constant, then there are many examples in Github) We have some IC, BC, and some measurements for `u(x, t)`, and we try to infer `u(x, t)` and `a(x, t)`. To solve this problem, and make it easy to explain, let us replace `u` with `u1` and `a` with `u2`, then we have the PDE `u1_t + u2 u1_x = 0`. We just assume we try to solve `u1` and `u2` from only one PDE with IC/BC for `u1` and some measurements for `u1`. Then this is similar to the example Lorenz_inverse.py, where we want to solve more than one functions. (Of course, you don't need the C1, C2, C3 in Lorenz example) DeepXDE dose not care how many PDE we have, how many IC/BC, etc. You just define all the information, and solve.

**Comment by axledos100** [comment]
Hello Lu Lu,

This makes sense! I've solved a few inverse problems using DEEPXDE in this manner. The approach is remarkable, and the results have been superb. I believe what I am asking is slightly different. My apologies for not being clear. Below I've attached an image of the general logic for which I'd like to use DEEPXDE/PINNs. The problem is for a simple diffusion equation. Is it possible to use DEEPXDE with this framework?

Best, 
Axle
![PINNs](https://user-images.githubusercontent.com/55243918/92131896-7b8e6280-edd4-11ea-9e00-260d281df95d.png)
